utils/helper.js

```diff
@@ -1,6 +1,7 @@
-import startCase from 'lodash.startcase'
+/* eslint-env node */
 
-import sidebar from '../sidebar'
+const startCase = require('lodash.startcase')
+const sidebar = require('./sidebar')
```

Comment:
Agree with Ivan, this file can be left in src/Documentation/ I think.

Reply:
Not sure if this is a good place for that either. It is basically docs stricture and is edited together withe docs, it is not a part of visual component.

Reply:
yep, I would say even helper should be part of the `src/Documentation`, not utils. To me `utils` is a place for library-like code that is used in a lot of places across the code - like array extensions, or string helpers, etc

Reply:
But we already use helper and sidebar data in multiple places:

- `server.js` to check if page exists,
- `pages/doc.js` to load content,
- `src/Documentation/Markdown/utils/remark-linker.js` to check if page exists,
- `src/Documentation/SidebarMenu/SidebarMenu.js` to draw sidebar. 

It's basically just docs structure and routing info, not just info for sidebar. So I think that utils is a good place for helper (but we probably should rename it). But for `sidebar.json` itself I think that it should be placed somewhere near the docs, because they are often edited together and without changing js code. What to you think?

Reply:
why don't you like idea of moving it top-level - to a separate directory like `sidebar`? both `helper` and `json`. It's just my perception of the `utils` is different. To me it's like `commons` - I would expect for stuff like I mentioned before - collections, strings, etc, not application-specific logic. I can imagine it will be confusing to others, so why don't we make it super explicit - `src\sidebar\sidebar.json` and `src\sidebar\helper.js` . It will make it super easy and clear to navigate, explain, document, etc.

Reply:
@shcheklein different reasons for `helper.js` and `sidebar.json` actually.

`sidebar.json` – I think that `src` folder is for application's source code and that data and content shouldn't be there. And I think that `sidebar.json` is a part of the content as it is mostly used as part of content editing tasks, and not programming tasks. So fixing something that is a part of the `src` folder to add a page to the docs didn't feel right to me.

`helper.js` – this file, on the other hand, should be placed inside the `src` folder, because it is a part of application source code, and I don't think that it should be a part of `SidebarMenu` component because it is used in a lot of other places. We can create a folder in `src` just for this file, but it looks wasteful for me to create a top-level folder just for it. (Also we already have a lot of folders there and no distinction between component folders and other folders, so for me, it is hard to navigate between them and look for specific folder right now)

P. S. Offtopic, but I think that we should move all react components to subfolder `components` to clean `src` a little.

Reply:
So, let's move `sidebar.json` to the very top at least, `utils` just don't make any sense for it. And `helper.js` - I'm fine with `utils` since it's dev facing anyways.

Reply:
we should update a few docs where sidebar.json is mentioned if we move it

Reply:
@shcheklein agreed. To the top of `src` folder or to the site's root folder?

Reply:
probably site's root? like you mentioned it's not the source code

Reply:
It can be part of `static/docs` indeed, close to the content that it defines. In this sense it's as _static_ as those Markdown files we have, so should be good.